packages/flutter/lib/src/widgets/editable_text_context_menu_button_items_builder.dart

```diff
@@ -0,0 +1,296 @@
+    'This feature was deprecated after v2.12.0-4.1.pre.',
+  )
+  static List<ContextMenuButtonItem>? buttonItemsForToolbarOptions(EditableTextState editableTextState, [TargetPlatform? targetPlatform]) {
+    final ToolbarOptions toolbarOptions = editableTextState.widget.toolbarOptions;
+    if (toolbarOptions == ToolbarOptions.empty) {
+      return null;
+    }
+    return <ContextMenuButtonItem>[
+      if (toolbarOptions.cut
+          && EditableTextContextMenuButtonItemsBuilder.canCut(editableTextState))
+        ContextMenuButtonItem(
+          onPressed: () {
+            editableTextState.selectAll(SelectionChangedCause.toolbar);
+          },
+          type: ContextMenuButtonType.selectAll,
+        ),
+      if (toolbarOptions.copy
+          && EditableTextContextMenuButtonItemsBuilder.canCopy(editableTextState))
+        ContextMenuButtonItem(
+          onPressed: () {
+            editableTextState.copySelection(SelectionChangedCause.toolbar);
+          },
+          type: ContextMenuButtonType.copy,
+        ),
+      if (toolbarOptions.paste && editableTextState.clipboardStatus != null
+          && EditableTextContextMenuButtonItemsBuilder.canPaste(editableTextState))
+        ContextMenuButtonItem(
+          onPressed: () {
+            editableTextState.pasteText(SelectionChangedCause.toolbar);
+          },
+          type: ContextMenuButtonType.paste,
+        ),
+      if (toolbarOptions.selectAll
+          && EditableTextContextMenuButtonItemsBuilder.canSelectAll(editableTextState, targetPlatform))
+        ContextMenuButtonItem(
+          onPressed: () {
+            editableTextState.selectAll(SelectionChangedCause.toolbar);
+          },
+          type: ContextMenuButtonType.selectAll,
+        ),
+    ];
+  }
+
+  @override
+  State<EditableTextContextMenuButtonItemsBuilder> createState() => _EditableTextContextMenuButtonItemsBuilderState();
+}
+
+class _EditableTextContextMenuButtonItemsBuilderState extends State<EditableTextContextMenuButtonItemsBuilder> with TickerProviderStateMixin {
+  bool get _canCut => EditableTextContextMenuButtonItemsBuilder.canCut(widget.editableTextState);
+
+  bool get _canCopy => EditableTextContextMenuButtonItemsBuilder.canCopy(widget.editableTextState);
+
+  bool get _canSelectAll => EditableTextContextMenuButtonItemsBuilder.canSelectAll(widget.editableTextState, widget.targetPlatform);
+
+  void _handleCut() {
+    return widget.editableTextState.cutSelection(SelectionChangedCause.toolbar);
+  }
+
+  void _handleCopy() {
+    return widget.editableTextState.copySelection(SelectionChangedCause.toolbar);
+  }
+
+  Future<void> _handlePaste() {
+    return widget.editableTextState.pasteText(SelectionChangedCause.toolbar);
+  }
+
+  void _handleSelectAll() {
+    return widget.editableTextState.selectAll(SelectionChangedCause.toolbar);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _ClipboardStatusBuilder(
```

Comment:
It can also change even if the system clipboard changes outside of the scope of EditableText or even Flutter, like copying some text in another app.

Reply:
No I meant _ClipboardStatusBuilder seems to be only useful to handle where the instance of widget.editableTextState.clipboardStatus are different from build to build. but it would only be different if the instance of editableTextState are different. so logic can be handled in _EditableTextContextMenuButtonItemsBuilderState.didUpdateWidget

I imagine you can just listen to widget.editableTextState.clipboardStatus directly in _EditableTextContextMenuButtonItemsBuilderState and update the listener in didUpdateWidget if the `widget.editableTextState != old.editableTextState`

Reply:
Ah I see.  Looks like it's probably cleaner that way, I'll change it 👍